typescript/sdk/src/core/read.ts

```diff
@@ -0,0 +1,54 @@
+import { providers } from 'ethers';
+
+import { Mailbox__factory } from '@hyperlane-xyz/core';
+import { Address, objMap, promiseObjAll } from '@hyperlane-xyz/utils';
+
+import { EvmHookReader } from '../hook/read.js';
+import { EvmIsmReader } from '../ism/read.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { ChainName } from '../types.js';
+
+import { CoreConfig } from './types.js';
+
+export class EvmCoreReader {
```

Comment:
Don't forget to export new things from the SDK's root index.ts and also to create a changeset :)

Reply:
done

Reply:
Changeset added